modules/campaigns/module.php

```diff
@@ -0,0 +1,123 @@
+<?php
+namespace Elementor\Modules\Campaigns;
+
+use Elementor\Core\Base\Module as BaseModule;
+use Elementor\Modules\LandingPages\Module as Landing_Page_Module;
+use Elementor\TemplateLibrary\Source_Local;
+use Elementor\Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Campaigns Module
+ *
+ * 'Campaigns' is a custom taxonomy that is associated with Popups, Landing Pages, and Form Submissions.
+ */
+class Module extends BaseModule {
+
+	public $campaigns_menu_slug;
+	private $terms;
+
+	public function get_name() {
+		return 'campaign';
+	}
+
+	private function get_taxonomy_page_slug() {
+		return 'edit-tags.php?taxonomy=' . $this->get_name() . '&post_type=page';
+	}
+
+	private function register_campaigns_taxonomy() {
+		$taxonomy_labels = [
+			'name' => _x( 'Campaigns', 'Campaigns taxonomy general name', 'elementor' ),
+			'singular_name' => _x( 'Campaign', 'Campaign singular name', 'elementor' ),
+			'search_items' => __( 'Search Campaigns', 'elementor' ),
+			'popular_items' => __( 'Popular Campaigns', 'elementor' ),
+			'all_items' => __( 'All Campaigns', 'elementor' ),
+			'edit_item' => __( 'Edit Campaign', 'elementor' ),
+			'update_item' => __( 'Update Campaign', 'elementor' ),
+			'add_new_item' => __( 'Add New Campaign', 'elementor' ),
+			'new_item_name' => __( 'New Campaign Name', 'elementor' ),
+			'separate_items_with_commas' => __( 'Separate Campaigns with commas', 'elementor' ),
+			'add_or_remove_items' => __( 'Add or remove Campaigns', 'elementor' ),
+			'choose_from_most_used' => __( 'Choose from the most used Campaigns', 'elementor' ),
+			'not_found' => __( 'No Campaigns found.', 'elementor' ),
+			'menu_name' => __( 'Campaigns', 'elementor' ),
+		];
+
+		$taxonomy_args = [
+			'labels' => $taxonomy_labels,
+			'hierarchical' => false,
+			'show_in_menu' => false,
+			'query_var' => is_admin(),
+			'rewrite' => false,
+		];
+
+		// Register taxonomy, associate it with Landing Pages and Popups (Actually 'elementor_library', not just Popups).
+		// Landing pages' CPT is just 'page'.
+		register_taxonomy( $this->get_name(), [ 'page', Source_Local::CPT ], $taxonomy_args );
```

Comment:
Add `TAXONOMY_NAME` constant with a value of `e-campaigns` and use it instead of `get_name()`

Reply:
Done.